src/runtime_src/core/common/drv/xrt_cu.c

```diff
@@ -136,6 +192,30 @@ int xrt_cu_thread(void *data)
 		spin_unlock_irqrestore(&xcu->pq_lock, flags);
 	}
 
+	if (!xcu->timeout)
```

Comment:
it might be problematic to detect CU timeout inside CU subdev. since it may just need more time to finish the command? you can't mark the board as bad unless both conditions below are true:
1. command is not returning from CU
2. user application does not want to wait any more and intent to leave (close)
if the application is happily waiting and CU command is being executed, the board should be in good state, right?

Reply:
If we would like to detect CU timeout, that should be CU subdevice. right? Since this is the only place to access CU registers and it knows status of commands.

Yes, a command might have a long execute time. We have "struct ert_init_kernel_cmd" command which could configure the timeout. If user know the CU needs more time. He could use ert_init_kernel_cmd to adjust timeout.

I agree that two conditions. My question is how to define "command is not returning from CU"?

I haven't support all of command types in ert.h(This is what I'm going to do next). After that, user could reconfigure the timeout of CU subdevice.

Reply:
After discussed with Max offline, when close context, the client should determine if CU hang or not. CU should only provide method not policy.